packages/api-reference/src/components/ApiReferenceLayout.vue

```diff
@@ -107,9 +108,15 @@ onBeforeMount(() => updateHash())
 const scrollToSection = async (id?: string) => {
   isIntersectionEnabled.value = false
   updateHash()
-
-  if (id) scrollToId(id)
-  else documentEl.value?.scrollTo(0, 0)
+  if (id) {
+    // Open the section if it's collapsed
+    const sectionId = getSectionId(id)
+    if (sectionId !== id && !isCollapsedSidebarItem(sectionId)) {
+      setCollapsedSidebarItem(getSectionId(id), true)
+      await sleep(100)
+    }
```

Comment:
```suggestion
      setCollapsedSidebarItem(getSectionId(id), true)
      await sleep(100)
```

tbh you can just set it to true, its just a hashmap

Reply:
We need to confirm sectionId !== id otherwise it triggers sometimes when it shouldn't